tests/validation/tests/v3_api/common.py

```diff
@@ -2735,3 +2735,113 @@ def get_node_details(cluster, client):
         if node.worker:
             break
     return client, node
+
+
+def create_service_account_configfile():
+    client, cluster = get_user_client_and_cluster()
+    create_kubeconfig(cluster)
+    name = random_name()
+    # create a service account
+    execute_kubectl_cmd(cmd="create sa {}".format(name), json_out=False)
+    # get the ca and token
+    res = execute_kubectl_cmd(cmd="get secret -o name", json_out=False)
+    secret_name = ""
+    for item in res.split("\n"):
+        if name in item:
+            secret_name = item.split("/")[1]
+            break
+    res = execute_kubectl_cmd(cmd="get secret {}".format(secret_name))
+    ca = res["data"]["ca.crt"]
+    token = res["data"]["token"]
+    token = base64.b64decode(token).decode()
+
+    # get the server ip, which is the control plane's ip
+    ip = os.environ.get('RANCHER_CONTROL_PLANE_IP', "None")
```

Comment:
Can we get this info from the `cluster` object? This is to be able to go end-to-end in Jenkins:

- Create cluster
- Get CP info
- Run tests
- Results
- Teardown

Reply:
good point. 
Now we get the ip from the control plane node.